core/adapters/multiply.go

```diff
@@ -2,33 +2,34 @@ package adapters
 
 import (
 	"encoding/json"
-	"math/big"
 
-	"chainlink/core/utils"
+	"github.com/shopspring/decimal"
 )
 
 // Multiply holds the a number to multiply the given value by.
 type Multiply struct {
-	Times *big.Float `json:"-"`
+	Times decimal.Decimal `json:"-"`
 }
 
 type jsonMultiply struct {
-	Times *utils.BigFloat `json:"times,omitempty"`
+	Times decimal.Decimal `json:"times,omitempty"`
```

Comment:
I'm surpised the tests are passing, but this breaks the API.  utils.BigFloat is used to accept both strings and json floats for decimal values

Reply:
It looks like `decimal.Decimal` appropriately handles strings and floats. Check out the new test cases I added. I'm open to alternatives if there are any downsides of `utils.BigFloat`? But the story did specifically request I replace the float types.